Make it possible to rename app passwords

kesselb opened the pull request:
Close #6717 
![peek 2019-01-20 21-02](https://user-images.githubusercontent.com/3902676/51444356-c5593080-1cf6-11e9-8cb5-3af5b18d7c51.gif)

Not sure if its save to use https://caniuse.com/#feat=keyboardevent-key here.

kesselb committed:
Extend regex to match DAVx5

https://gitlab.com/bitfireAT/davx5-ose/commit/7ce739c271f9b4f9b3cd7c16c526a774a7d99bdc
Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>

kesselb committed:
Clipboard -> ClipboardJS

https://github.com/nextcloud/server/issues/12915
Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>

kesselb committed:
Make app password description editable

Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>

kesselb committed:
Add canRename & name to tests

Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>

ChristophWurst commented on settings/Controller/AuthSettingsController.php:
```diff
@@ -203,6 +206,11 @@ public function update($id, array $scope) {
 			'filesystem' => $scope['filesystem']
 		]);
+
+		if ($token instanceof DefaultToken || $token instanceof PublicKeyToken) {
```
May I suggest to introduce a new interface for this, like a `INamedToken extends IToken` that has a getter and setter for the name? That would make the code easier to follow and makes the API more clear :+1: 

Of course this can be an *internal* interface in `OC` as it's not used outside of the server.

ChristophWurst reviewed (dismissed):
Just a nitpick regarding API design. Looks great otherwise! Thanks for adding this feature!

kesselb committed:
Add interface INamedToken

Remove $token instanceof DefaultToken || $token instanceof PublicKeyToken

Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>

jancborchardt reviewed (dismissed):
Very nice work! :+1: Only a nitpick from me too: Could you modify it so the height of the row doesn’t change in rename mode? :)

kesselb committed:
Improve usability

- Trigger rename by doubleclick on token name
- Use action rename to open and close
- Line height stays the same

Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>

kesselb committed:
Don't update token if current session token

Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>

danielkesselberg commented:
![peek 2019-02-02 14-51](https://user-images.githubusercontent.com/3902676/52164951-16572480-26fa-11e9-827b-77920bcc3aea.gif)

danielkesselberg commented:
https://drone.nextcloud.com/nextcloud/server/15458/168 :boom: 

> Argument 3 passed to OC\\Settings\\Controller\\AuthSettingsController::update() must be of the type string

There are two ways to fix this:

1. Adjust test and add name parameter to request
2. Make name optional

Are desktop or mobile using this endpoints? If so we should make name optional.

kesselb committed:
Mock name for integration test

Signed-off-by: Daniel Kesselberg <mail@danielkesselberg.de>

danielkesselberg commented:
Integration-Auth is happy again. Time for another review ;-)

MorrisJobke reviewed (approved):
Tested and works 👍

rullzer commented:
@danielkesselberg as a followup could you also add activities for renaming a token?

rullzer commented:
Ah and I just forgot. Renaming should probably also require a password confirmation...
